.github/actions/setup-binary/action.yml

```diff
@@ -0,0 +1,159 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Setup Binary'
+
+description: |-
+  Use this action to install and cache a binary for use within a GitHub workflow.
+
+inputs:
+  name:
+    description: 'The name of the binary.'
+    required: true
+  download_url:
+    description: 'The download URL for the binary.'
+    required: true
+  install_dir:
+    description: 'The directory to install the binary.'
+    default: '.'
+    required: false
+  compression_method:
+    description: 'The compression method used on the downloaded artifact. Allowed values: tar, zip, tgz'
+    default: ''
+    required: false
+  enable_cache:
+    description: 'Enable the use of caching for the binary.'
+    default: 'false'
+    required: false
+  override_cache_key:
+    description: 'Override the default value of the cache key.'
+    required: false
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Validate required inputs'
+      shell: 'bash'
+      env:
+        NAME: '${{ inputs.name }}'
+        DOWNLOAD_URL: '${{ inputs.download_url }}'
+      run: |
+        errors=false
+
+        if [ -z "${NAME}" ]; then
+          errors=true
+          echo "::error ::name is a required input"
+        fi
+
+        if [ -z "${DOWNLOAD_URL}" ]; then
+          errors=true
+          echo "::error ::download_url is a required input"
+        fi
+
+        if [ "$errors" = true ]; then
+          exit 1
+        fi
+
+    - name: 'Validate compression method'
+      shell: 'bash'
+      if: |
+        !contains(fromJSON('["tar", "zip", "tgz", ""]'), inputs.compression_method)
+      env:
+        COMPRESSION_METHOD: '${{ inputs.compression_method }}'
+      run: |
+        echo "::error ::invalid compression_method '${COMPRESSION_METHOD}', allowed values: tar, zip, tgz"
+        exit 1
+
+    - name: 'Cache path'
```

Comment:
I wonder if you could skip this step and instead pass it directly to the cache:
```
id: 'cache-binary'
...
with: 
  key: '${{ inputs. override_cache_key || format("{0}-{1}-{2}", OS, ARCH, inputs.name) }}'
```

Reply:
ack